src/DI/PhpReflection.php

```diff
@@ -59,13 +59,19 @@ public static function getDeclaringClass(\ReflectionProperty $prop)
 	 */
 	public static function getPropertyType(\ReflectionParameter $prop)
 	{
-		try {
-			return ($ref = $prop->getClass()) ? $ref->getName() : NULL;
-		} catch (\ReflectionException $e) {
-			if (preg_match('#Class (.+) does not exist#', $e->getMessage(), $m)) {
-				return $m[1];
+		if (PHP_VERSION_ID >= 70000) {
+			return (string) $prop->getType();
+		} elseif ($prop->isArray()) {
```

Comment:
`callable` is used here https://github.com/pavelkouril/di/commit/6649b249e4b4b3e812e1c5baef689f843e6596a0#diff-49f23e184ca8ecc097b83d102bfdc2a0R90, so PR is not needed, but ok, I'll prepare support for 2.3/master.